cdci_data_analysis/flask_app/dispatcher_query.py

```diff
@@ -322,39 +320,62 @@ def __init__(self, app,
         logger.info("constructed %s:%s for data_server_call_back=%s", self.__class__, self, data_server_call_back)
 
     @staticmethod
-    def inspect_state(app):
+    def free_up_space(app):
         token = request.args.get('token', None)
-        if token is None:
-            # TODO what about using the RequestNotAuthorized exception?
-            # raise RequestNotAuthorized("A token must be provided.")
-            return make_response('A token must be provided.'), 403
-        try:
-            secret_key = app.config.get('conf').secret_key
-            decoded_token = tokenHelper.get_decoded_token(token, secret_key)
-            logger.info("==> token %s", decoded_token)
-        except jwt.exceptions.ExpiredSignatureError:
-            # raise RequestNotAuthorized("The token provided is expired.")
-            return make_response('The token provided is expired.'), 403
-        except jwt.exceptions.InvalidTokenError:
-            # raise RequestNotAuthorized("The token provided is not valid.")
-            return make_response('The token provided is not valid.'), 403
 
-        recent_days = request.args.get('recent_days', 3, type=float)
-        job_id = request.args.get('job_id', None)
+        app_config = app.config.get('conf')
+        secret_key = app_config.secret_key
 
-        roles = tokenHelper.get_token_roles(decoded_token)
+        output, output_code = tokenHelper.validate_token_from_request(token=token, secret_key=secret_key,
+                                                                      required_roles=['job manager'],
+                                                                      action="free_up space on the server")
 
-        required_roles = ['job manager']
-        # no need to have both, one of those two is sufficient
-        if not any(item in roles for item in required_roles):
-            lacking_roles = ", ".join(sorted(list(set(required_roles) - set(roles))))
-            message = (
-                f"Unfortunately, your privileges are not sufficient for this type of request.\n"
-                f"Your privilege roles include {roles}, but the following roles are missing: {lacking_roles}."
-            )
-            # raise RequestNotAuthorized(message=message)
-            return make_response(message), 403
+        if output_code is not None:
+            return make_response(output, output_code)
+
+        numb_folders_to_delete = int(request.args.get('folder_to_delete', 5))
```

Comment:
this is not a very useful input, since it can only be provided meaningfully if the existing number of folders is known, and their usage is considered. I'd rather suggest "max_age" - to protect very recent folders. Also, we can go in the folders and check if their job is finished.

Reply:
i considered using the number of weeks to refer t the minimum age a certain folder should have in order to be deleted.

If it's "old enough" it will also check the status of its job, and if its done then it will be deleted.

Does it make sense? Is the number of weeks a good enough way to decide on the deletion ?

Reply:
in principle this is good, just one more element: if there is a token, we could check if it has expired, and only then delete.

note however that number of weeks is not the right scale since we are rather talking about days here.

Reply:
> if it has expired, and only then delete.
> 

Which advantages does it bring?

> note however that number of weeks is not the right scale since we are rather talking about days here.
>
That is why I was asking

Reply:
> > if it has expired, and only then delete. 
> Which advantages does it bring?

We are using these records to reconstruct request parameters by jobid. These records should be available as long as the token provided with them is valid.

Reply:
by the way, this way if we detect incomplete jobs older than the soft limit, we can send ourselves an alert. since this is not right

Reply:
good idea, I just introduced those additional checks

BTW testing will be tricky

Reply:
> good idea, I just introduced those additional checks
> 
> BTW testing will be tricky

you can set mtime/ctime in the past

https://docs.python.org/3/library/os.html#os.utime